libs/features/mas/src/merch-card.js

```diff
@@ -363,6 +363,16 @@ export class MerchCard extends LitElement {
     }
 
     async checkReady() {
+        const timeoutPromise = new Promise((resolve) =>
```

Comment:
btw, checkReady, when used externally, should not dispatch events. Would be nice if you add it in your PR.